packages/@uppy/provider-views/src/ProviderView/ProviderView.tsx

```diff
@@ -157,6 +162,16 @@ export default class ProviderView<M extends Meta, B extends Body> {
       this.provider.provider,
       this.provider,
     )
+
+    // Configure debounced search with test override
+    const testHookSymbol = Symbol.for('uppy test: searchDebounceMs')
+    const testWait = (
+      ProviderView as unknown as Record<symbol, number | undefined>
+    )[testHookSymbol]
+    const wait = testWait ?? 500
+    const debounceOpts =
+      testWait === 0 ? { leading: true, trailing: true } : undefined
+    this.#searchDebounced = debounce(this.#search, wait, debounceOpts)
```

Comment:
One last question, why can't the test function without this custom debounce time?

Reply:
@Murderlon  see my comments above, if we sleep 600ms everywhere it's going to slow down our tests severely, and sleeping in tests is not good practice (can lead to race conditions and flakes)

Reply:
I'll take a 600ms slowdown anytime over introducing all this conditional test code inside our library code. But I'm a bit late with bringing this up. I'll leave it for you @qxprakash to decide. Either way you can merge.

Reply:
We had very flakey e2e tests before. let's not repeat that mistake. setTimeout inside tests is not robust.